### ISSUE TITLE
grpo 训练过程中，出现自定义的模型 template 没有调用

### ISSUE THREAD

**Issue by WarranWeng** [modelscope/ms-swift#6006]
**Describe the bug**
在 grpo 训练过程中，出现自定义的模型 template 没有调用的问题。命令如下：
swift rlhf \
    --rlhf_type grpo \
    --model model_ckpt \
    --external_plugins scripts/train_qwen_vl/RFT/plugin.py \
    --custom_register_path data/dataset_register.py model/Qwen_2_5_vl_custom/model.py \
    --reward_funcs my_format \
    --template qwen2_5_vl_custom \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host 127.0.0.1 \
    --vllm_server_port 8001 \
    --train_type full \
    --torch_dtype bfloat16 \
    --dataset video-my \
    --load_from_cache_file false \
    --max_completion_length 1024 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-6 \
    --gradient_accumulation_steps 2 \
    --save_strategy 'steps' \
    --eval_strategy 'steps' \
    --eval_steps 1000 \
    --save_steps 1000 \
    --save_total_limit 10 \
    --logging_steps 1 \
    --output_dir model/out \
    --warmup_ratio 0.01 \
    --dataloader_num_workers 4 \
    --num_generations 2 \
    --temperature 1.0 \
    --deepspeed zero3 \
    --log_completions true \
    --num_iterations 1 \
    --async_generate false \
    --beta 0.001 \

自定义的 model template 是对 encode 函数重构：
class Qwen2_5VLTemplateCustom(Qwen2_5VLTemplate):

    @torch.inference_mode()
    def encode(self,
               inputs: Union[TemplateInputs, Dict[str, Any], InferRequest],
               return_template_inputs: bool = False,
               return_length: bool = False) -> Dict[str, Any]:
        """The entrance method of Template!

        Returns:
            return {'input_ids': List[int], 'labels': Optional[List[int]], ...}
        """
        if self.is_training:
            inputs = self._get_message(inputs, is_train=True)

        return super().encode(inputs, return_template_inputs, return_length)
其中：_get_message函数是返回带有 message 的字典。
出现报错如下：

<img width="787" height="318" alt="Image" src="https://github.com/user-attachments/assets/0eadd4da-b0a5-4995-9e1d-37671e3a97f2" />

这里输入 batch 没有过上面的自定义的 model template，导致 message 字段缺失。

非 grpo 训练上述代码是可以跑通的，自定义的 model template 也正常调用。

请问如何解决？

**Comment by hjh0119** [comment]
please do not override `encode`, override `_encode` instead
